### PR TITLE
Configurable HTTP compression.

### PR DESCRIPTION
**Pull request by gianm**
Based on https://github.com/metamx/http-client/pull/26.

### CODE REVIEW

**Comment by fjy** [comment]
👍

File: docs/content/configuration/broker.md [hunk 1]
```diff
@@ -37,7 +37,8 @@ Druid uses Jetty to serve HTTP requests.
 |`druid.server.http.numThreads`|Number of threads for HTTP requests.|10|
 |`druid.server.http.maxIdleTime`|The Jetty max idle time for a connection.|PT5m|
 |`druid.broker.http.numConnections`|Size of connection pool for the Broker to connect to historical and real-time nodes. If there are more queries than this number that all need to speak to the same node, then they will queue up.|20|
-|`druid.broker.http.readTimeout`|The timeout for data reads.|PT15M|
+|`druid.broker.http.compressionCodec`|Compression codec the Broker uses to communicate with historical and real-time nodes. May be "gzip" or "identity".|gzip|
```
Comment (fjy) [inline docs/content/configuration/broker.md]: real-time processes

File: server/src/main/java/io/druid/guice/http/DruidHttpClientConfig.java [hunk 2]
```diff
@@ -30,6 +30,8 @@
 
 public class DruidHttpClientConfig
 {
+  private final String DEFAULT_COMPRESSION_CODEC = "gzip";
```
Comment (fjy) [inline server/src/main/java/io/druid/guice/http/DruidHttpClientConfig.java]: i think uncompressed shuld be default

Comment (gianm) [inline server/src/main/java/io/druid/guice/http/DruidHttpClientConfig.java]: I don't want to change the default in this PR.

File: docs/content/configuration/broker.md (hunk 1 above)
Comment (gianm) [inline docs/content/configuration/broker.md]: Changed.

**Comment by jon-wei** [comment]
:+1:

**Comment by jon-wei** [comment]
PR looks good, I do have a suggestion but it would involve releasing another version of http-client:

The String->CompressionCodec enum conversion could be done within http-client, e.g.:

```
    @JsonCreator
    public static CompressionCodec fromString(String name)
    {
      return valueOf(name.toUpperCase());
    }
```

Then DruidHttpClientConfig could use CompressionCodec enums directly instead of accepting String

**Pull request merged**